### PR TITLE
Add new game option to randomly select from map options.

### PR DESCRIPTION
**Pull request by FiretronP75**

### CODE REVIEW

**Comment by FiretronP75** [comment]
I couldn't figure out how to give the `isPortrait` info to the `MapParametersTable` so the new `MultiCheckBoxTable` does not actually add extra padding for portrait. Not sure if it really needs it. I don't know how to test portrait mode. I just borrowed that bit from `ModCheckBoxTable`

**Comment by yairm210** [comment]
You can check portrait mode on PC by setting the window much higher than it is wide

Also, pics please of how it looks when before/active

**Comment by FiretronP75** [comment]
Thanks. In portrait mode, it looks like the extra padding is not needed, so I'll just cut that useless code out.

**Comment by FiretronP75** [comment]
Tall Before
Tall Collapsed
Tall Expanded
Wide Normal
Wide Collapsed
ed](https://user-images.githubusercontent.com/2472969/210263517-37153e17-29f1-4115-9dbd-f4cbbf9f52b4.png)
2ae524e18.png)

**Comment by FiretronP75** [comment]
Portrait:
Tall Before
Tall Collapsed
Tall Expanded

**Comment by FiretronP75** [comment]
Landscape:
Wide Normal
Wide Collapsed
Wide Expanded
Wide Expanded More

**Comment by FiretronP75** [comment]
The 3 toggle buttons (no ancient ruins, no natural wonders, worlld wrap) look like they should have a bar on top of them now to make them more different than the Resource Settings above it.

**Comment by FiretronP75** [comment]
The Map Type drops down changes the view back and forth correctly when you select Generated or Random Generated.

**Comment by FiretronP75** [comment]
They all start expanded, not collapsed.

**Comment by FiretronP75** [comment]
> The 3 toggle buttons (no ancient ruins, no natural wonders, worlld wrap) look like they should have a bar on top of them now to make them more different than the Resource Settings above it.

I did this.

Collapsable Other Settings

**Comment by FiretronP75** [comment]
They now all start collapsed, as that is more user friendly, especially in portrait view.

**Comment by FiretronP75** [comment]
I would like to get the toggle buttons all left aligned but nothing is working.

**Comment by FiretronP75** [comment]
New screen shot after label changes.
Enabled Settings

File: core/src/com/unciv/ui/newgamescreen/MapParametersTable.kt [hunk 1]
```diff
@@ -293,14 +293,25 @@ class MapParametersTable(
...
-        }).colspan(2).center().row()
-        add("World wrap maps are very memory intensive - creating large world wrap maps on Android can lead to crashes!"
-            .toLabel(fontSize = 14).apply { wrap=true }).colspan(2).fillX().row()
+        if (mapParameters.type === MapType.randomGenerated) {
+            add(ExpanderTab("{Other Settings}:", persistenceID = "NewGameOtherSettings") {
+                it.defaults().pad(5f,0f)
+                it.addNoRuinsCheckbox()
+                it.addNoNaturalWondersCheckbox()
+                it.addWorldWrapCheckbox()
+                it.add("World wrap maps are very memory intensive - creating large world wrap maps on Android can lead to crashes!"
```
Comment (yairm210) [inline core/src/com/unciv/ui/newgamescreen/MapParametersTable.kt:302]: Define this in one place, so if it's changed is changed once

File: android/assets/jsons/translations/template.properties [hunk 2]
```diff
@@ -345,9 +345,12 @@ Map file =
 Max Turns = 
 Could not load map! = 
 Generated = 
+Random Generated = 
+Which options should be available to the random selection?
```
Comment (yairm210) [inline android/assets/jsons/translations/template.properties:349]: Missing =

**Pull request merged**